lib/server.go

```diff
@@ -2010,9 +2168,30 @@ func (srv *Server) _handleBlock(pp *Peer, blk *MsgDeSoBlock) {
 		return
 	}
 
-	// If we get here, it means we're in SyncStateFullySynced, which is great.
+	// If we get here, it means we're in SyncStateFullyCurrent, which is great.
 	// In this case we shoot a MEMPOOL message over to the peer to bootstrap the mempool.
 	srv._maybeRequestSync(pp)
+
+	///////////////////// PoS Validator Consensus Initialization /////////////////////
+
+	// Exit early if the chain isn't SyncStateFullyCurrent.
+	if srv.blockchain.chainState() != SyncStateFullyCurrent {
+		return
+	}
+
+	// Exit early if the current tip height is below the PoS cutover height. We are ready to enable
+	// the PoS validator FastHotStuffConsensus once we reach the final block of the PoW protocol.
+	// This requires the block height check to use ProofOfStake2ConsensusCutoverBlockHeight-1
+	tipHeight := uint64(srv.blockchain.blockTip().Height)
+	if tipHeight < srv.blockchain.GetFinalPoWBlockHeight() {
```

Comment:
Why not `!srv.blockchain.IsPoSBlockHeight` here?

Reply:
We need to start the FastHotStuffConsensus once the chain has connected the final PoW block, which has a height of `ProofOfStake2ConsensusCutoverBlockHeight - 1`. The final PoW block is the genesis block for the PoS chain, on top of which the PoS consensus votes or times out on

Reply:
Clarified the comment further in PR https://github.com/deso-protocol/core/pull/1010